fast math test overhaul

andykaylor opened the pull request:
- Fix -fno-unsafe-math-optimizations behavior
- [WIP] Major change to fast-math.c lit test
- Clean up the checks in the fast-math driver test

arsenm reviewed (approved):
I think this works for the test. I'm slightly confused by the PR not-stacking with the test changes on top of the other PR

zahiraam reviewed (approved):
LGTM. Thanks.

andykaylor commented:
> I think this works for the test. I'm slightly confused by the PR not-stacking with the test changes on top of the other PR

That's user error on my part. There are commits here that I didn't intend to be part of this PR. I'll clean it up before I submit it.

Commit:
Clean up the checks in the fast-math driver test

This change replaces most of the cc1 option checks in the driver test
for fast-math option handling. These changes rely on the assumption that
the order in which the driver emits floating-point options is stable.

The changes also rely on the assumption that the order of prefixes
listed on the FileCheck command line is unimportant and that all
prefixed checks will be combined and checked as if they were a single
prefix. At the time of the change, that worked.
